### PR TITLE
Clarify on-prem vs. cloud instructions for GitHub Enterprise

### PR DESCRIPTION
**Pull request by cstavitsky**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Resolves https://github.com/getsentry/sentry-docs/issues/8665

### CODE REVIEW

File: docs/product/integrations/source-code-mgmt/github/index.mdx
Comment (vivianyentran) [inline docs/product/integrations/source-code-mgmt/github/index.mdx:44]:
```suggestion
If you're using GitHub Enterprise Cloud, follow the [instructions for GitHub](/product/integrations/source-code-mgmt/github/#install).
```

File: docs/product/integrations/source-code-mgmt/github/index.mdx
Comment (vivianyentran) [inline docs/product/integrations/source-code-mgmt/github/index.mdx:39]: If you update the header text, you'll need to update the link in line 11 so that it goes to the right anchor

**Review by vivianyentran — APPROVED** [review]

File: docs/product/integrations/source-code-mgmt/github/index.mdx
Comment (cstavitsky) [reply docs/product/integrations/source-code-mgmt/github/index.mdx:39]: Oh didn't even think about that (and how that might break the existing links for GHE already out there...! I'll undo that change and leave the header as-is. Good feedback 👍

**Comment by vivianyentran** [comment]
Just wanted to follow up on this PR as it's still in draft form. Wondering if you're waiting on anything else before merging this?

**Comment by cstavitsky** [comment]
> Just wanted to follow up on this PR as it's still in draft form. Wondering if you're waiting on anything else before merging this?

Yes, going to see if an engineer who works on integrations can give this a review as well. Following up on that now, will promote from draft to a real PR, but please hold off on merging til I can get an approval there as well

**Review by schew2381 — APPROVED** [review]
lgtm

**Comment by cstavitsky** [comment]
@vivianyentran I think we're good to get this merged 👍 Thanks again @schew2381 for the review!

**Comment by vivianyentran** [comment]
sounds good! will go ahead and merge this in!

**Pull request merged**
